### ISSUE TITLE
Implement ternary operator support

### ISSUE THREAD

**Issue by alexdovzhanyn** [alexdovzhanyn/ThetaLang#24]
Currently we support control flow operations such as if / else if / else. You can assign variables to the result of control flow operations like so:

```theta
x<String> = if (1 == 1) 'Yes' else 'No'
```

This works fine, but it would be nice to add some syntactic sugar so we could also do:

```theta
x<String> = 1 == 1 ? 'Yes' : 'No'
```

In order to allow for this, we need to update the parser to recognize this as a control flow node.

- [ ] Update the lexer to allow for `?` tokens
- [ ] Update the parser to turn ternary operations into `ControlFlowNode`s
- [ ] Add tests to the lexer and parser to verify the changes

**Comment by syfuuu** [comment]
@alexdovzhanyn I can work on this. Please assign this to me.

**Comment by alexdovzhanyn** [comment]
@syfuuu You got it!

**Comment by syfuuu** [comment]
Hey, Are you sure `x<String> = if (1 == 1) 'Yes' else 'No'` works? its giving me segmentation fault. I already worked on lexer but parser is but tricky. Try to understand how it works incase of if/else. Is there any communication channel for discussing things related to this project?

**Comment by alexdovzhanyn** [comment]
@syfuuu Yeah it should work, though there may be a bug. It looks like currently `if (1 == 1) 'Yes' else 'No'` gets parsed correctly, however it fails if you try to assign it to a variable like `x<String> = if (1 == 1) 'Yes' else 'No'`. I'll make a separate bug ticket for that. But for now you can build the ternary based on `if (1 == 1) 'Yes' else 'No'`.

We have a discord channel [here](https://discord.gg/mzWgbhGQ6C), feel free to hop in.

**Comment by alexdovzhanyn** [comment]
@syfuuu I've made another issue to track that bug here: https://github.com/alexdovzhanyn/ThetaLang/issues/37
